### PR TITLE
Clean-up parameter settings and display only axis/stepper supported by the machine

### LINKED ISSUE #1883: TFT70 BTT stealtchop X and Y not showed in TFT mode
Issue occurred updating to latest firmware TFT70:
After updating TFT 70 to latest firmware in the menu->options->printer->parameters->TMC stealtchop  for X and Y is showed value 0.000 even if stealtchop is enabled in marlin (2.0.8). Is possible set to 1.00 but after exit 0.000 is showed. In marlin mode the stealtchop is correcly showed as enabled. Same issue not occurred updating with the same version of firmware a TFT35V3.

### PR DESCRIPTION
**Pull request by guruathwal**
- Clean-up parameter settings.
- Display only axis/stepper supported by the machine.
- Fix some bug in parameter parsing in parseACK.c and interfaceCmd.c
- Workaround to display stealthchop settings if not reported by Marlin firmware when it is turned off.
- Separate persistent info code from parameter settings.

Build report:

Device | Old RAM Usage | New RAM usage | RAM Change | Old Flash Size | New Flash Size | Flash Size Reduced
-- | -- | -- | -- | -- | -- | --
TFT35 V1.1 | 17544 | 17596 | 52 | 148188 | 147608 | -580
TFT35 V1.2 | 17544 | 17596 | 52 | 148068 | 147480 | -588
TFT35 V2 | 18288 | 18344 | 56 | 157472 | 156876 | -596
TFT35 V3.0 | 18712 | 18764 | 52 | 175740 | 175144 | -596
TFT35 E3 V3.0 | 18716 | 18768 | 52 | 176412 | 175800 | -612
TFT50 V3.0 | 18716 | 18768 | 52 | 176416 | 175812 | -604
TFT70 V3.0 | 18844 | 18896 | 52 | 177572 | 176976 | -596
TFT28 V1.0 | 17544 | 17596 | 52 | 148228 | 147640 | -588
TFT28 V3.0 | 18716 | 18768 | 52 | 176244 | 175648 | -596
TFT24 V1.1 | 18688 | 18740 | 52 | 171716 | 171136 | -580
MKS TFT32 V1.3 | 18624 | 18676 | 52 | 171280 | 170676 | -604
MKS TFT32 V1.4 | 18624 | 18676 | 52 | 171728 | 171124 | -604

Resolves: #1671 , Fixes #1883

### CODE REVIEW

File: TFT/src/User/API/HomeOffsetControl.c [hunk 1]
```diff
@@ -48,8 +48,7 @@ float homeOffsetSetValue(float value)
 // Get current Z offset value
 float homeOffsetGetValue(void)
 {
-  z_offset_value = getParameter(P_HOME_OFFSET, Z_STEPPER);
-
+  z_offset_value = getParameter(P_HOME_OFFSET, AXIS_INDEX_Z);
```
Comment (digant73) [inline TFT/src/User/API/HomeOffsetControl.c:51]: please maintain the empty line before "return". All my code use that syntax (as for the variable definition etc..)

**Comment by digant73** [comment]
split "reseolves  #1671 , #1883" to two lines. #1883 is not currently marked as fixed by this PR

File: TFT/src/User/API/MachineParameters.c [hunk 2]
```diff
@@ -1,90 +1,172 @@
...
+  {"S%.0f\n",            "S1 T0 D%.2f\n", "S1 T1 D%.2f\n", NULL,           NULL,         NULL,           NULL,           NULL},           // Filament Diameter (Enable, E0, E1)
+  {"X%.0f\n",            "I1 X%.0f\n",    "Y%.0f\n",       "I1 Y%.0f\n",   "Z%.0f\n",    "I1 Z%.0f\n",   "T0 E%.0f\n",   "T1 E%.0f\n"},   // Current (X, X2, Y, Y2, Z, Z2, E0, E1)
+  {"X%.0f\n",            "I1 X%.0f\n",    "Y%.0f\n",       "I1 Y%.0f\n",   "Z%.0f\n",    "I1 Z%.0f\n",   NULL,           NULL},           // bump Sensitivity (X, X2, Y, Y2, Z, Z2)
+  {"X%.0f\n",            "I1 X%.0f\n",    "Y%.0f\n",       "I1 Y%.0f\n",   "Z%.0f\n",    "I1 Z%.0f\n",   "T0 E%.0f\n",   "T1 E%.0f\n"},   // TMC Hybrid Threshold Speed (X, X2, Y, Y2, Z, Z2, E0, E1)
+  {"S%.0f X\n",          "S%.0f I1 X\n",  "S%.0f Y\n",     "S%.0f I1 Y\n", "S%.0f Z\n",  "S%.0f I1 Z\n", "S%.0f T0 E\n", "S%.0f T1 E\n"}, // TMC StealthChop (X, X2, Y, Y2, Z, Z2, E0, E1)
+  {"S4 Z%.2f\nG29 S0\n", NULL,            NULL,            NULL,           NULL,         NULL,           NULL,           NULL},           // MBL offset
+};
+
+const VAL_TYPE parameterValType[PARAMETERS_COUNT][MAX_ELEMENT_COUNT] = {
+
```
Comment (digant73) [inline TFT/src/User/API/MachineParameters.c:76]: maybe the blank line has to be removed

File: TFT/src/User/Menu/ParameterSettings.c [hunk 3]
```diff
@@ -30,111 +29,110 @@ const LABEL parameterTypes[PARAMETERS_COUNT] = {
 };
 
 const LISTITEM eepromItems[P_SETTINGS_COUNT] = {
-  // icon           ItemType   Item Title              item value text(only for custom value)
-  {CHARICON_SAVE,  LIST_LABEL, LABEL_SETTINGS_SAVE,    LABEL_BACKGROUND},
-  {CHARICON_UNDO,  LIST_LABEL, LABEL_SETTINGS_RESTORE, LABEL_BACKGROUND},
-  {CHARICON_RESET, LIST_LABEL, LABEL_SETTINGS_RESET,   LABEL_BACKGROUND},
+  // icon            ItemType    Item Title              item value text(only for custom value)
+    {CHARICON_SAVE,  LIST_LABEL, LABEL_SETTINGS_SAVE,    LABEL_BACKGROUND},
```
Comment (digant73) [inline TFT/src/User/Menu/ParameterSettings.c:33]: too much indentation?

Comment (guruathwal) [reply TFT/src/User/Menu/ParameterSettings.c]: ok now?

Comment (digant73) [reply TFT/src/User/Menu/ParameterSettings.c]: yes. Two chars of indentation

**Pull request merged**